apps/api/src/routers/admin.py

```diff
@@ -210,6 +210,61 @@ async def confirm_attendance() -> None:
         )
 
 
+@router.get(
+    "/attending",
+    dependencies=[
+        Depends(require_role([Role.DIRECTOR, Role.ORGANIZER, Role.VOLUNTEER]))
+    ],
+)
+async def attending() -> list[dict[str, object]]:
+    """Fetch all applicants who have a status of ATTENDING"""
+    records = await mongodb_handler.retrieve(
+        Collection.USERS,
+        {"role": Role.APPLICANT, "status": Status.ATTENDING},
+        [
+            "_id",
+            "status",
+            "role",
+            "application_data.first_name",
+            "application_data.last_name",
+        ],
+    )
+
+    return records
+
+
+@router.post("/checkin/{uid}")
+async def checkin(
+    uid: str,
+    associate: User = Depends(
+        require_role([Role.DIRECTOR, Role.ORGANIZER, Role.VOLUNTEER])
+    ),
+) -> None:
+    """Check-in applicant at IrvineHacks"""
+    record: Optional[dict[str, object]] = await mongodb_handler.retrieve_one(
+        Collection.USERS, {"_id": uid, "role": Role.APPLICANT}
+    )
+    if not record:
+        raise HTTPException(status.HTTP_404_NOT_FOUND)
+    elif record["status"] != Status.ATTENDING:
+        raise RuntimeError("Applicant status must be ATTENDING")
+
+    new_checkin_entry = {"uid": associate.uid, "time": utc_now()}
+
+    update_status = await mongodb_handler.raw_update_one(
+        Collection.USERS,
+        {"_id": uid},
+        {
+            "$push": {"checkins": new_checkin_entry},
+        },
+    )
+
+    if not update_status:
+        raise HTTPException(status.HTTP_404_NOT_FOUND)
```

Comment:
Issue: 404 isn't exactly the best status code to raise here. If we get to this point, then the user must exist in the database and has a status of `ATTENDING`, so if `raw_update_one` returns a falsy value, that means something is seriously wrong. A 500 would therefore be better.